fv3core/stencils/delnflux.py

```diff
@@ -1,218 +1,1108 @@
-from gt4py.gtscript import PARALLEL, computation, interval
+from typing import Optional
+
+import gt4py.gtscript as gtscript
+import numpy as np
+from gt4py.gtscript import (
+    __INLINED,
+    FORWARD,
+    PARALLEL,
+    computation,
+    horizontal,
+    interval,
+    region,
+)
 
 import fv3core._config as spec
-import fv3core.utils.corners as corners
 import fv3core.utils.gt4py_utils as utils
-from fv3core.decorators import gtstencil
-from fv3core.stencils.basic_operations import copy_stencil
-from fv3core.utils.typing import FloatField, FloatFieldIJ
+from fv3core.decorators import FrozenStencil, gtstencil
+from fv3core.utils.grid import axis_offsets
+from fv3core.utils.typing import FloatField, FloatFieldIJ, FloatFieldK
 
 
-@gtstencil
-def fx2_order(q: FloatField, del6_v: FloatFieldIJ, fx2: FloatField, order: int):
-    with computation(PARALLEL), interval(...):
-        fx2[0, 0, 0] = del6_v * (q[-1, 0, 0] - q)
-        fx2[0, 0, 0] = -1.0 * fx2 if order > 1 else fx2
+def copy_corners_x_nord0(q: FloatField):
```

Comment:
given these are copies of copy_corners except for the interval, you could consider calling copy_corners with different k origins and domains (make an object for each in init, or something). But this solution is closer to the one intended to be implemented, where we did need to replicate the copy_corner logic to introduce the if nord > 0 condition. So I am ok with leaving this as-is, just wanting to bring up that point again.

Reply:
Thank you @rheacangeo, we can give that a try.